plugins/org.yakindu.sct.model.stext/src/org/yakindu/sct/model/stext/validation/STextJavaValidator.java

```diff
@@ -409,78 +464,81 @@ public void checkValueReferenedBeforeDefined(Scope scope) {
 		Set<QualifiedName> defined = Sets.newHashSet();
 		for (Declaration declaration : declarations) {
-			if (declaration instanceof VariableDefinition) {
-				VariableDefinition definition = (VariableDefinition) declaration;
-				if (!definition.isConst())
-					return;
-				Expression initialValue = definition.getInitialValue();
-				List<Expression> toCheck = Lists.newArrayList(initialValue);
-				TreeIterator<EObject> eAllContents = initialValue.eAllContents();
-				while (eAllContents.hasNext()) {
-					EObject next = eAllContents.next();
-					if (next instanceof Expression)
-						toCheck.add((Expression) next);
-				}
-				for (Expression expression : toCheck) {
-					EObject referencedObject = null;
-					if (expression instanceof FeatureCall)
-						referencedObject = ((FeatureCall) expression).getFeature();
-					else if (expression instanceof ElementReferenceExpression)
-						referencedObject = ((ElementReferenceExpression) expression).getReference();
-					if (referencedObject instanceof VariableDefinition) {
-						if (!defined.contains(nameProvider.getFullyQualifiedName(referencedObject)))
-							error(REFERENCE_CONSTANT_BEFORE_DEFINED, definition,
-									TypesPackage.Literals.PROPERTY__INITIAL_VALUE);
-					}
+			if (!(declaration instanceof VariableDefinition)) {
+				continue;
+			}
+			VariableDefinition definition = (VariableDefinition) declaration;
+			if (!definition.isConst())
+				return;
+			Expression initialValue = definition.getInitialValue();
+			List<Expression> toCheck = Lists.newArrayList(initialValue);
+			TreeIterator<EObject> eAllContents = initialValue.eAllContents();
+			while (eAllContents.hasNext()) {
+				EObject next = eAllContents.next();
+				if (next instanceof Expression)
+					toCheck.add((Expression) next);
+			}
+			for (Expression expression : toCheck) {
+				EObject referencedObject = null;
+				if (expression instanceof FeatureCall)
+					referencedObject = ((FeatureCall) expression).getFeature();
+				else if (expression instanceof ElementReferenceExpression)
+					referencedObject = ((ElementReferenceExpression) expression).getReference();
+				if (referencedObject instanceof VariableDefinition) {
+					if (!defined.contains(nameProvider.getFullyQualifiedName(referencedObject)))
+						error(REFERENCE_CONSTANT_BEFORE_DEFINED, definition,
+								TypesPackage.Literals.PROPERTY__INITIAL_VALUE);
 				}
-				defined.add(nameProvider.getFullyQualifiedName(definition));
 			}
+			defined.add(nameProvider.getFullyQualifiedName(definition));
 		}
 	}
 
 	@Check(CheckType.NORMAL)
 	public void checkUnusedExit(final Exit exit) {
-		if (exit.getParentRegion().getComposite() instanceof org.yakindu.sct.model.sgraph.State
-				&& exit.getOutgoingTransitions().isEmpty()) {
-			org.yakindu.sct.model.sgraph.State state = (org.yakindu.sct.model.sgraph.State) exit.getParentRegion()
-					.getComposite();
+		if (!(exit.getParentRegion().getComposite() instanceof org.yakindu.sct.model.sgraph.State)
```

Comment:
Same as above - old version was more readable in my opinion because now the condition contains more parenthesis, even wrapping multiple lines.